remove-http.php

```diff
@@ -62,38 +84,89 @@ public function option() {
-                    $pieces = explode( ':', strtolower( $header ) );
-                    $content_type = trim( $pieces[1] );
-                    break;
-                }
+    private function is_applicable_content_type() {
+        # Try to find the value of the Content-Type header and check if it is text/html
+        foreach ( headers_list() as $header ) {
+            if ( stripos( $header, 'content-type:' ) === 0 ) {
+                $pieces = explode( ':', $header );
+                return strpos( strtolower( trim( $pieces[1] ) ), 'text/html' ) === 0;
             }
-            # If the content-type is 'NULL' or 'text/html', apply rewrite
-            if ( is_null( $content_type ) || substr( $content_type, 0, 9 ) === 'text/html' ) {
-                $tag = 'script|link|base|img|form|a|meta|iframe|svg|div';
-                $attribute = 'href|src|srcset|action|content|data-project-file';
-                # If 'Protocol Relative URL' option is checked, only apply change to internal links
-                if ( $this->option == 1 ) {
-                    # Remove protocol from home URL
-                    $website = preg_replace( '/https?:\/\//', '', home_url() );
-                    # Remove protocol from internal links
-                    $links = preg_replace( '/(<(' . $tag . ')([^>]*)(' . $attribute . ')=["\'])https?:\/\/' . $website . '/i', '$1//' . $website, $links );
-                }
-                # Else, remove protocols from all links
-                else {
-                    $links = preg_replace( '/(<(' . $tag . ')([^>]*)(' . $attribute . ')=["\'])https?:\/\//i', '$1//', $links );
-                }
+        }
+
+        # We didn't find it, it was not declared so assume HTML
+        return true;
+    }
+
+    private function process_element_attributes( DOMElement $element, array $attributes, $replace_regex ) {
+        foreach ( $attributes as $attribute ) {
+            # Check that the element actually has the target attribute
+            if ( !$element->hasAttribute( $attribute ) ) {
+                continue;
             }
-            # Return protocol relative links
-            return $links;
-        } );
+
+            # Get the current attribute value, modify it and set it back on the element
+            $content = $element->getAttribute( $attribute );
+            $result = preg_replace( $replace_regex, '$1', $content );
+            $element->setAttribute( $attribute, $result );
+        }
+    }
+
+    private function process_html($html ) {
+        # Check that we have DOM loaded
+        if ( !class_exists( 'DOMDocument' ) || !class_exists( 'DOMXPath' ) ) {
+            throw new Fact_Maven_Remove_HTTP_Processing_Failed_Exception;
+        }
+
+        # Try to create a document and xpath
+        $doc = new \DOMDocument();
+        if ( !@$doc->loadHTML( $html, LIBXML_HTML_NODEFDTD ) ) {
+            throw new Fact_Maven_Remove_HTTP_Processing_Failed_Exception;
```

Comment:
Here the exception is thrown when the DOM parse operation fails. This could happen if the HTML is very malformed, or if the content is not actually HTML.